src/compiler/checker.ts

```diff
@@ -12684,8 +12679,17 @@ namespace ts {
                             }
                             break;
                         case SyntaxKind.FunctionExpression:
-                            if ((<FunctionExpression>location).name) {
-                                copySymbol(location.symbol, meaning);
+                        case SyntaxKind.ClassExpression:
+                            // The reason we are not using copySymbol for function expression and class expression
+                            // is that copySymbol will not copy a symbol into SymbolTable if the symbol has name prefix
+                            // with "__". Therefore, if class expression or function expression have declared name,
+                            // we will add the symbol into the table using its declared name

```

Comment:
Why not change the logic of `copySymbol`?

Also, the comment for `copySymbol` says `// Returns 'true' if we should stop processing symbols.`, but it returns `void`.